controllers/servingruntime_controller.go

```diff
@@ -509,25 +509,69 @@ func (r *ServingRuntimeReconciler) SetupWithManager(mgr ctrl.Manager,
 }
 
 func (r *ServingRuntimeReconciler) requestsForRuntimes(namespace string,
-	filter func(*kserveapi.ServingRuntime) bool) []reconcile.Request {
+	filter func(string) bool) []reconcile.Request {
 	var opts []client.ListOption
 	if namespace != "" {
 		opts = []client.ListOption{client.InNamespace(namespace)}
 	}
-	list := &kserveapi.ServingRuntimeList{}
-	if err := r.Client.List(context.TODO(), list, opts...); err != nil {
+	runtimes := &kserveapi.ServingRuntimeList{}
+	if err := r.Client.List(context.TODO(), runtimes, opts...); err != nil {
 		r.Log.Error(err, "Error listing ServingRuntimes to reconcile", "namespace", namespace)
 		return []reconcile.Request{}
 	}
-	requests := make([]reconcile.Request, 0, len(list.Items))
-	for i := range list.Items {
-		rt := &list.Items[i]
-		if filter == nil || filter(rt) {
-			requests = append(requests, reconcile.Request{
-				NamespacedName: types.NamespacedName{Name: rt.Name, Namespace: rt.Namespace},
-			})
+
+	var requests []reconcile.Request
+	if r.EnableCSRWatch {
```

Comment:
I just realized it would be better here to also go to the `else` case if there are no CSRs.
i.e. something like
```go
var csrs *kserveapi.ClusterServingRuntimeList
if r.EnableCSRWatch {
    csrs = &kserveapi.ClusterServingRuntimeList{}
    if err := r.Client.List(context.TODO(), csrs); err != nil {
    	r.Log.Error(err, "Error listing ClusterServingRuntimes to reconcile")
	return []reconcile.Request{}
    } 
}
if csrs != nil && len(csrs.Items) > 0 {
    srnns := make(map[types.NamespacedName]struct{})
    // ...
```